kernel/fatal.c

```diff
@@ -0,0 +1,92 @@
+#include <kernel.h>
+#include <kernel_internal.h>
+#include <misc/printk.h>
+#include <arch/cpu.h>
+#include <logging/log_ctrl.h>
+#include <fatal.h>
+
+/* LCOV_EXCL_START */
+FUNC_NORETURN __weak void z_arch_system_halt(unsigned int reason)
+{
+	ARG_UNUSED(reason);
+
+	for (;;) {
+		k_cpu_idle();
+	}
+}
+/* LCOV_EXCL_STOP */
+
+/* LCOV_EXCL_START */
+__weak void k_sys_fatal_error_handler(unsigned int reason,
+				      const NANO_ESF *esf)
+{
+	ARG_UNUSED(esf);
+
+	printk("Halting system.\n");
+	(void)irq_lock();
+
+	z_arch_system_halt(reason);
+}
+/* LCOV_EXCL_STOP */
+
+void z_fatal_error(unsigned int reason, const NANO_ESF *esf)
```

Comment:
Should probably take the time to rename and document the "NANO_ESF" type.  Maybe "struct z_arch_excinfo" or something?

Also, IIRC most of the existing architectures pass a garbage/fallback value here (c.f. the "0xdeaddead" IP, etc...).  For situations where you can't synthesize a proper frame (most architectures are just lazy I think) we probably want a portable way to do that.

Reply:
agreed on all points, working on it.